src/main/java/dk/bec/bookanything/controller/FacilityController.java

```diff
@@ -0,0 +1,130 @@
+    }
+
+    @GetMapping("/facilities/{id}/features")
+    ResponseEntity<List<FeatureReadDto>> getFeaturesForFacility(@PathVariable("id") Long id) {
+        return facilityService.getFeaturesForFacility(id).map(featureReadDtos -> new ResponseEntity<>(featureReadDtos, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
+
+    @GetMapping("facilities/{id}/days_open")
+    ResponseEntity<List<DayOpenReadDto>>  getDaysOpenForFacility(@PathVariable("id") Long id) {
+        return facilityService.getDaysOpenForFacility(id).map(dayOpenReadDtos -> new ResponseEntity<>(dayOpenReadDtos, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
+    }
+
+    @PostMapping("/facilities")
+    ResponseEntity<FacilityReadDto> createFacility(@Valid @RequestBody FacilityCreateDto facilityDto) {
+        FacilityEntity facility = facilityMapper.mapFacilityCreateDtoToEntity(facilityDto, null);
+        Optional<FacilityEntity> facilityOptional = facilityService.createFacility(facility);
+
+        return facilityOptional.map(facilityEntity -> new ResponseEntity<>(facilityMapper.mapFacilityEntityToReadDto(facilityEntity), HttpStatus.OK)).orElseGet(() -> new ResponseEntity<>(HttpStatus.BAD_REQUEST));
+
+    }
+
+    @PutMapping("/facilities/{id}")
+    ResponseEntity<FacilityReadDto> updateFacility(@PathVariable("id") Long id, @Valid @RequestBody FacilityCreateDto facilityDto) {
+        FacilityEntity facility = facilityMapper.mapFacilityCreateDtoToEntity(facilityDto, id);
+        Optional<FacilityEntity> facilityOptional = facilityService.updateFacility(id, facility);
+
+        return facilityOptional.map(facilityEntity -> new ResponseEntity<>(facilityMapper.mapFacilityEntityToReadDto(facilityEntity), HttpStatus.OK)).orElseGet(() -> new ResponseEntity<>(HttpStatus.BAD_REQUEST));
+    }
+
+    @DeleteMapping("/facilities/{id}")
+    ResponseEntity<FacilityCreateDto> deleteFacility(@PathVariable("id") Long id) {
+        facilityService.deleteFacilityById(id);
+
+        return facilityService.getFacilityById(id).isPresent() ? new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR) : new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @GetMapping("/facilities")
+    ResponseEntity<List<FacilityReadDto>> getFacilitiesByCityAndType(@RequestParam(name = "facilityTypeId", required = false) Long facilityTypeId,
+                                                              @RequestParam(name = "city", required = false) String city) {
+        if (facilityTypeId != null && city != null) {
+            Optional<FacilityTypeEntity> facilityTypeById = Optional.ofNullable(facilityTypeService.getFacilityTypeById(facilityTypeId));
+            Optional<List<AddressEntity>> addressEntitiesByCity = addressService.getAddressesByCity(city);
+            if (facilityTypeById.isPresent() && addressEntitiesByCity.isPresent()) {
```

Comment:
Sure, but what if this statement is in fact false? That would mean the following:
1. 89 was executed
2. 90 was executed
3. 96 will be executed 
4. 102 will executed
Conclusion: Too many calls to datasbase..